pkg/controller/propagator/placementDecisionMapper.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package propagator
+
+import (
+	"context"
+
+	clusterv1alpha1 "github.com/open-cluster-management/governance-policy-propagator/pkg/apis/cluster/v1alpha1"
+	policiesv1 "github.com/open-cluster-management/governance-policy-propagator/pkg/apis/policy/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type placementDecisionMapper struct {
+	client.Client
+}
+
+func (mapper *placementDecisionMapper) Map(obj handler.MapObject) []reconcile.Request {
+	object := obj.Meta
+	log.Info("Reconcile Request for PlacementDecision %s in namespace %s", object.GetName(), object.GetNamespace())
+
+	// get the placement name from the placementdecision
+	placementName := object.GetLabels()["cluster.open-cluster-management.io/placement"]
+	if len(placementName) == 0 {
+		return nil
+	}
+
+	pbList := &policiesv1.PlacementBindingList{}
+	// find pb in the same namespace of placementrule
+	err := mapper.List(context.TODO(), pbList, &client.ListOptions{Namespace: object.GetNamespace()})
```

Comment:
Could you also add a field selector for at least the PlacementRef name? Filtering on the server side will reduce network IO consumption and should be faster.

Reply:
yes, will try this out